### PR TITLE
Move hard coded parameters to runconfing

### PR DESCRIPTION
**Pull request by oberonia78**
This PR introduces an update to the `initial_threshold.py` script, addressing the previously hard-coded values used to define the threshold range. In the original implementation, any computed threshold falling outside these predefined limits was disregarded, potentially limiting the adaptability and effectiveness of the thresholding process across varied datasets.

By enabling user-defined threshold ranges and extensive testing to identify optimal settings, it aims to improve performance, in particular, along the water boundary.

### CODE REVIEW

File: src/dswx_sar/initial_threshold.py [hunk 1]
```diff
@@ -1039,7 +1059,8 @@ def determine_threshold(
 
             if threshold > tau_bound_gauss:
                 threshold = tau_bound_gauss
-                idx_tau_bound_gauss = np.searchsorted(intensity_bins, threshold)
+                idx_tau_bound_gauss = np.searchsorted(intensity_bins,
+                                                      threshold)
                 tri_lowmaxind_cands, _ = find_peaks(intensity_counts[0:idx_tau_bound_gauss+1], distance=5)
                 if not tri_lowmaxind_cands.any():
```
Comment (seongsujeong) [inline src/dswx_sar/initial_threshold.py:1065]: Please consider splitting the lone  lines here.

**Review by seongsujeong — APPROVED** [review]
LGTM. Pending minor suggestions.

File: src/dswx_sar/initial_threshold.py [hunk 2]
```diff
@@ -1564,8 +1599,46 @@ def save_threshold_dict(threshold, block_row, block_col):
...
+                  cfg,
+                  winsize=200,
+                  thres_max=None):
+    """
+    Process sub-blocks of SAR intensity data for water detection based on
+    the specified configuration.
+
+    Parameters
+    ----------
+    intensity : ndarray
```
Comment (LiangJYu) [inline src/dswx_sar/initial_threshold.py:1625]:
```suggestion
    intensity : np.array
```

File: src/dswx_sar/initial_threshold.py [hunk 3]
```diff
@@ -1564,8 +1599,46 @@ def save_threshold_dict(threshold, block_row, block_col):
...
+        SAR intensity data, can be 2D or 3D (polarizations).
+    water_body_subset : ndarray
```
Comment (LiangJYu) [inline src/dswx_sar/initial_threshold.py:1627]:
```suggestion
    water_body_subset : np.array
```

**Review by LiangJYu — APPROVED** [review]
LGTM. Just a couple nits.

**Pull request merged**